runners/direct-java/src/main/java/org/apache/beam/runners/direct/TransformExecutor.java

```diff
@@ -82,11 +88,14 @@ private TransformExecutor(
     this.onComplete = completionCallback;
 
     this.transformEvaluationState = transformEvaluationState;
+    this.context = context;
     this.thread = new AtomicReference<>();
   }
 
   @Override
   public void run() {
+    MetricsContainer metricsContainer = new MetricsContainer(transform.getFullName());
```

Comment:
We want a separate container for each bundle and each step. This would require the `setMetricsContainer` taking all that information, and holding a map to expose the containers for all steps in a bundle, etc. I think this is going to be better handled as part of the individual runners (which already have the bundle/step information) and/or the Fn API.